### PR TITLE
Refactor CMake scripts to automate server build with one step

### PR DESCRIPTION
**Pull request by vfreex**
The new CMake build scripts allow you to use cmake to build all server components.
This change gets rid of the original awkward and faulty `.sh` build scripts.

Refer to `server/README.md` for build instructions.

@shiywang @shokakucarrier PTAL

### CODE REVIEW

**Comment by shiywang** [comment]
@vfreex this is great!! I kind of forget but previous team talk server side technically can run on MacOS/Linux/Windows... Are we gonna cramp the support only for Linux?

File: server/src/db_proxy_server/business/UserModel.cpp [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #include "../DBPool.h"
 #include "../CachePool.h"
 #include "Common.h"
-#include "SyncCenter.h"
+#include "../SyncCenter.h"
```
Comment (shiywang) [inline server/src/db_proxy_server/business/UserModel.cpp:15]: I did not have a thorough look of what `.sh` script did But it seems bump some `.h` files into this folder and now we put it into a shared folder?

Comment (vfreex) [reply server/src/db_proxy_server/business/UserModel.cpp:15]: No, those build `.sh` scripts don't copy common headers, but there are duplicates in different directories (I guess they must be copied manually before committing to Git).

I removed as many duplicates as I can and tweaked the include directories in the CMakeLists.txt files to pass the compilation.

**Comment by vfreex** [comment]
> @vfreex this is great!! I kind of forget but previous team talk server side technically can run on MacOS/Linux/Windows... Are we gonna cramp the support only for Linux?

I didn't test compilation of the original TeamTalk on Windows or Mac. But it would be better if the server is cross-platform, especially on Windows Server!

Probably I am going to try it on my Windows desktop.

**Pull request merged**